### DISCUSSION TITLE
after wire:model reload lose svg icon

### DISCUSSION THREAD

**Discussion by marty12321** [livewire/livewire#3620] [7. Bugs (archived)]
I'm using BS5 and icons in input field:
`<div class="mb-3 filled form-group tooltip-end-top">
<svg xmlns="http://www.w3.org/2000/svg" width="20" height="20" ...</path></svg>
<input class="form-control" placeholder="email" type="email" id="email" wire:model.defer="email" name="email" wtx-context="75E93530-57DB-471A-9302-1F586882F204">
                                            </div>`

after submit and reload failed validation, I have <i> element instead of svg
`<div class="mb-3 filled form-group tooltip-end-top">
                        <i data-cs-icon="email"></i>
                        <input class="form-control" placeholder="email" type="email" id="email" wire:model.defer="email" name="email" wtx-context="00EEF9E3-6A9A-4267-854B-4F14818821C5">
                                            </div>`

how can I prevent from this?

**Comment by ryatkins** [comment]
Quick note, use 3 back ticks on the top and bottom of your code to make them more readable code blocks.  Single tick to do inline code on a sentence. 

Livewire seems to be replacing your code because it has changed in some way.   From your sample, I can't tell if you have an outer `div` wrapping your code to help with the DOM diffing that happens.

Is wtx-context supposed to be changing between requests?  

I see your svg replaced with `<i data-cs-icon="email"></i>` - is that piece of code someone in your blade file?

Can you show more of your blade and Livewire component code?

**Reply by marty12321** [reply]
this the blade part, after submit i lose the svg and get <i ...> back, I think this could be some changes after rendering with JS, trying to find the part in the template, I've bought this template and it works with plain Laravel, but not with livewire https://themeforest.net/item/acorn-bootstrap-html-laravel-net-admin-template/32360322
```  <div>
                <form id="loginForm"  wire:submit.prevent="loginUser" class="tooltip-end-bottom" novalidate>
                    @csrf

                    <div class="mb-3 filled form-group tooltip-end-top">
                        <i data-cs-icon="email"></i>
                        <input class="form-control" placeholder="email" type="email" id="email" wire:model.defer="email" name="email"/>
                        @error('email')
                        <span class="invalid-feedback" role="alert">
                                        <strong>{{ $message }}</strong>
                                    </span>
                        @enderror
                    </div>
                    <div class="mb-3 filled form-group tooltip-end-top">
                        <i data-cs-icon="lock-off"></i>
                        <input class="form-control pe-7" type="password" class="form-control  @error('password') is-invalid @enderror"
                               id="password" wire:model.defer="password" name="password" placeholder="hasło"/>
                        <a class="text-small position-absolute t-3 e-3"
                           href="{{ url('/Pages/Authentication/ForgotPassword') }}">Zapomniałeś?</a>
                        @error('password')
                        <span class="invalid-feedback" role="alert">
                                        <strong>{{ $message }}</strong>
                                    </span>
                        @enderror
                    </div>
                    <button type="submit" class="btn btn-lg btn-primary">Zaloguj</button>
                </form>
            </div>```

**Reply by matt-h** [reply]
It looks like your HTML there does use the `i` tag in that spot. You must have some javascript code that replaces the `i` tag with SVG after page load.

To get that to work you'd have to have it also call that library code to run after livewire updates its DOM.
```
<script>
    document.addEventListener("DOMContentLoaded", () => {
        Livewire.hook('element.updated', (el, component) => {
              // TODO: Call whatever code you need to replace the i tags with SVGs.
       })
    });
</script>
```

**Comment by ryatkins** [comment]
@marty12321 

I would limit your component to the simplest test possible and then build up from there

```
<div>
  <form id="loginForm"  wire:submit.prevent="loginUser"
     <input class="form-control" placeholder="email" type="email" id="email" wire:model.defer="email" name="email" />
     <button type="submit" class="btn btn-lg btn-primary">Zaloguj</button>
  </form>
</div>
```

I'm not sure it matters, but I added a space between your closing slash and your last parameter in your input tag above. `name="email" />`

**Reply by marty12321** [reply]
yes, I found the problem, it is a https://lineicons.com/ fonts and js script with csicons.replace() function to make svg from <i> element.
Now, I'm looking for best way to invoke this function after response from livewire component.

**Reply by ryatkins** [reply]
Can you replace your "i" tag icons and make them into Blade components?  

Seems like this would be the better choice for the long run.
